src/main/java/use_cases/turn_end_use_case/TurnEndInteractor.java

```diff
@@ -0,0 +1,111 @@
+
+public class TurnEndInteractor implements TurnEndInputBoundary {
+
+    private final GameState gameState;
+    private final TurnEndOutputBoundary turnEndPresenter;
+
+    public TurnEndInteractor(GameState gameState, TurnEndOutputBoundary turnEndPresenter) {
+        this.gameState = gameState;
+        this.turnEndPresenter = turnEndPresenter;
+    }
+    
+    @Override
+    public void passTurn() {
+        gameState.switchPlayer();
+        // 1. pass player's name to presenter
+        // 2. whatever other effects that are needed
+    }
+
+    @Override
+    public TurnEndResponseModel triggerEndTurnEffects() {
+        Player currentPlayer = gameState.getCurrentPlayer();
+        Player nextPlayer = gameState.getOpposingPlayer();
+
+
+        // // 1. Check if the current player has a structure in play
+        // // 1a. Yes, lets trigger the event
+        if (currentPlayer.getStructure() != null) {
+            // Trigger the effects needed
+            StructureCard currentStructure = currentPlayer.getStructure();
+            List<CreatureCard> allTargets = new ArrayList<>();
+
+            switch (currentStructure.getTargetType()) {
+                // on the player themselves
+                case SELF:
+                    for (CardEffect effect : currentStructure.getEffects()) {
+                        if (effect instanceof PlayerStatsEffect) {
+                            ((PlayerStatsEffect) effect).invokeEffect(currentPlayer);
+                        }
+                    }
+                    break;
+                case ENEMY:
+                    allTargets.addAll(nextPlayer.getCreatures());
+                    break;
+                case ALL:
+                    allTargets.addAll(nextPlayer.getCreatures());
+                    allTargets.addAll(currentPlayer.getCreatures());
+                    break;
+                case FRIENDLY:
+                    allTargets.addAll(currentPlayer.getCreatures());
+                    break;
+                default:
+                    break;
+            }
+
+            for (CreatureCard creature : allTargets) {
+                for (CardEffect effect : currentStructure.getEffects()) {
+                    if (effect instanceof CreatureStatsEffect) {
+                        ((CreatureStatsEffect) effect).invokeEffect(creature);
+                    }
+                }
+            }
+        }
+
+        List<Integer> creatureIds1 = new ArrayList<>();
+        List<Integer> creatureIds2 = new ArrayList<>();
+        List<Integer> hitPoints1 = new ArrayList<>();
+        List<Integer> hitPoints2 = new ArrayList<>();
+        List<Integer> attack1 = new ArrayList<>();
+        List<Integer> attack2 = new ArrayList<>();
+        List<Integer> playerHandIds1 = new ArrayList<>();
+        List<Integer> playerHandIds2 = new ArrayList<>();
+        
+        for (CreatureCard creature : currentPlayer.getCreatures()) {
+            creature.clearStun();
```

Comment:
Is it normal to clearStun at the end of the turn?

Reply:
I'll just make a comment here, actually we were discussing this earlier. 
I'd say for this one it's the game mechanics deciding what we do here, and for now it's reasonable to have the Stun Effect cleared, consider this case:
- Player 1 stuns an opponent creature, that's Player 2's creature
- Player 2 begins the turn, during the turn, the stunned creature cannot move.
- After Player 2 ends their turn, the stunned creature goes back to normal.
If we were to clear the stun at the start of turn, then it would be:
- Player 1 stuns that creature
- Player 2 begins the turn, and in that moment the stun is cleared
- Player 2 can now use that creature for attacking, and the stun was pointless.
So for now we'll have the clearStun() happen here.

Reply:
Awesome, thanks  for the clarifcation!